packages/core/src/utils/file-utils.ts

```diff
@@ -78,18 +77,44 @@ export function symlinkJunction(to: string, from: string) {
  * returns the first occurrence of the node_modules, or null of none could be found.
  * @param basePath starting point
  */
-export async function findNodeModules(basePath: string): Promise<string | null> {
+export async function findNodeModulesList(
+  basePath: string
+): Promise<{
+  nodeModulesList: string[];
+  basePath: string;
+}> {
   basePath = path.resolve(basePath);
-  const nodeModules = path.resolve(basePath, 'node_modules');
-  try {
-    await fs.promises.stat(nodeModules);
-    return nodeModules;
-  } catch (e) {
-    const parent = path.dirname(basePath);
-    if (parent === basePath) {
-      return null;
-    } else {
-      return findNodeModules(path.dirname(basePath));
+  const nodeModulesList: string[] = [];
+
+  const dirBfsQueue: string[] = (await fs.promises.readdir(basePath)) ?? [];
+
+  while (dirBfsQueue.length > 0) {
+    const dir = dirBfsQueue.pop();
+
+    if (!dir) {
+      continue;
     }
+
+    if (dir && path.basename(dir) === '.stryker-tmps') {
```

Comment:
I don't think this exception is needed. 

First of all, the default stryker-tmp name is `.stryker-tmp` (minus the `s`) and second of all, it is overridable by the user (for example `--tempDirName tmp`). But I don't think it is really needed, as long as you make sure to not symlink symlinks.

Reply:
Fixed, using the this.options.tempDirName